src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -766,151 +361,53 @@ private class PowerShellResult
-                fileCorrections = new Dictionary<string, MarkerCorrection>();
-                newEntryNeeded = true;
-            }
+            CorrectionTableEntry fileCorrections = _mostRecentCorrectionsByFile.GetOrAdd(
+                scriptFile.DocumentUri,
+                CorrectionTableEntry.CreateForFile);
 
-            fileLock.Wait();
-            try
+            fileCorrections.Corrections.Clear();
+
+            for (int i = 0; i < markers.Count; i++)
             {
-                if (newEntryNeeded)
-                {
-                    // If we create a new entry, we should do it after acquiring the lock we just created
-                    // to ensure a competing thread can never acquire it first and read invalid information from it
-                    _mostRecentCorrectionsByFile[scriptFile.DocumentUri] = (fileLock, fileCorrections);
-                }
-                else
-                {
-                    // Otherwise we need to clear the stale corrections
-                    fileCorrections.Clear();
-                }
+                ScriptFileMarker marker = markers[i];
 
-                foreach (ScriptFileMarker marker in markers)
-                {
-                    // Does the marker contain a correction?
-                    Diagnostic markerDiagnostic = GetDiagnosticFromMarker(marker);
-                    if (marker.Correction != null)
-                    {
-                        string diagnosticId = GetUniqueIdFromDiagnostic(markerDiagnostic);
-                        fileCorrections[diagnosticId] = marker.Correction;
-                    }
+                Diagnostic diagnostic = GetDiagnosticFromMarker(marker);
 
-                    diagnostics.Add(markerDiagnostic);
+                if (marker.Correction != null)
+                {
+                    string diagnosticId = GetUniqueIdFromDiagnostic(diagnostic);
+                    fileCorrections.Corrections[diagnosticId] = marker.Correction;
                 }
-            }
-            finally
-            {
-                fileLock.Release();
+
+                diagnostics[i] = diagnostic;
             }
 
-            // Always send syntax and semantic errors.  We want to
-            // make sure no out-of-date markers are being displayed.
-            _languageServer.Document.PublishDiagnostics(new PublishDiagnosticsParams()
+            _languageServer.Document.PublishDiagnostics(new PublishDiagnosticsParams
             {
                 Uri = new Uri(scriptFile.DocumentUri),
-                Diagnostics = new Container<Diagnostic>(diagnostics),
+                Diagnostics = new Container<Diagnostic>(diagnostics)
             });
         }
 
-        public async Task<IReadOnlyDictionary<string, MarkerCorrection>> GetMostRecentCodeActionsForFileAsync(string documentUri)
+        private static ConcurrentDictionary<string, MarkerCorrection> CreateFileCorrectionsEntry(string fileUri)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'CreateFileCorrectionsEntry'.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4961943)